From d289049cf550b975487f94dfbd7af9d5a50b2233 Mon Sep 17 00:00:00 2001
From: Roland McGrath <roland@hack.frob.com>
Date: Fri, 2 Sep 2016 16:57:59 -0700
Subject: [PATCH] Fix generic wait3 after union wait_status removal.

	* sysdeps/posix/wait3.c: Don't treat STAT_LOC as a union, since it's
	not any more.

(cherry picked from commit 9a3d16ac152447567bfc822497c564a0630c79fe)
---
 ChangeLog             | 5 +++++
 sysdeps/posix/wait3.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 01272446091e..2738295ab10d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2016-09-02  Roland McGrath  <roland@hack.frob.com>
 
+	* sysdeps/posix/wait3.c: Don't treat STAT_LOC as a union, since it's
+	not any more.
+
+2016-09-02  Roland McGrath  <roland@hack.frob.com>
+
 	* sysdeps/nacl/clock.c (clock): nacl_abi_clock_t -> nacl_irt_clock_t
 
 2016-08-17  Florian Weimer  <fweimer@redhat.com>
diff --git a/sysdeps/posix/wait3.c b/sysdeps/posix/wait3.c
index cf43d973a7b7..73722d2be652 100644
--- a/sysdeps/posix/wait3.c
+++ b/sysdeps/posix/wait3.c
@@ -33,7 +33,7 @@ __wait3 (int *stat_loc, int options, struct rusage *usage)
       __set_errno (ENOSYS);
       return (pid_t) -1;
     }
-  return __waitpid (WAIT_ANY, stat_loc.__iptr, options);
+  return __waitpid (WAIT_ANY, stat_loc, options);
 }
 
 weak_alias (__wait3, wait3)
-- 
2.10.2

